src/mon/FSCommands.cc

```diff
@@ -424,6 +424,32 @@ class SetHandler : public FileSystemCommandHandler
           fs->mds_map.set_balancer(val);
         });
       return true;
+    } else if (var == "bal_rank_mask") {
+      string bin_string;
+      if (val.empty()) {
+        ss << "unsetting the metadata balancer rank mask";
+	return -EINVAL;
+      } else {
+	int r = parse_hex(val, bin_string, MAX_MDS, ss);
```

Comment:
@batrick The validation and translation logic is moved here and simplified. Would you please take a loot at?